lib/vision/index.js

```diff
@@ -0,0 +1,1376 @@
+            safeSearchAnnotation: Vision.formatSafeSearchAnnotation_,
+            textAnnotations: Vision.formatEntityAnnotation_
+          };
+
+          return formatMethodMap[type](annotation, options);
+        };
+      }
+
+      var originalResp = extend(true, {}, resp);
+
+      var detections = images
+        .map(function() {
+          // Group detections by image...
+          //
+          //   detections = [
+          //     // Image one:
+          //     [
+          //       {
+          //         faceAnnotations: {},
+          //         labelAnnotations: {},
+          //         ...
+          //       }
+          //     ],
+          //
+          //     // Image two:
+          //     [
+          //       {
+          //         faceAnnotations: {},
+          //         labelAnnotations: {},
+          //         ...
+          //       }
+          //     ]
+          //   ]
+          return annotations.splice(0, types.length);
+        })
+        .map(mergeArrayOfObjects)
+        .map(function(annotations) {
+          if (Object.keys(annotations).length === 0) {
+            // No annotations found, represent as an empty result set.
+            return [];
+          }
+
+          for (var annotationType in annotations) {
+            if (annotations.hasOwnProperty(annotationType)) {
+              var annotationGroup = arrify(annotations[annotationType]);
+
+              var formattedAnnotationGroup = annotationGroup
+                .map(formatAnnotationBuilder(annotationType));
+
+              // An annotation can be singular, e.g. SafeSearch. It is either
+              // violent or not. Unlike face detection, where there can be
+              // multiple results.
+              //
+              // Be sure the original type (object or array) is preserved and
+              // not wrapped in an array if it wasn't originally.
+              if (!is.array(annotations[annotationType])) {
+                formattedAnnotationGroup = formattedAnnotationGroup[0];
+              }
+
+              delete annotations[annotationType];
+              var strippedAnnotationType = annotationType
+                .replace(/Ann.*$/, '')
+                .replace('imageProperties', 'properties');
+              annotations[strippedAnnotationType] = formattedAnnotationGroup;
+            }
+          }
+
+          if (types.length === 1) {
+            // Only a single detection type was asked for, so no need to box in
+            // the results. Make them accessible without using a key.
+            var key = Object.keys(annotations)[0];
+            annotations = annotations[key];
+          }
+
+          return annotations;
+        });
+
+      // If only a single image was given, expose it from the array.
+      callback(null, isSingleImage ? detections[0] : detections, originalResp);
+    });
+  });
+};
+
+/**
+ * Run face detection against an image.
+ *
+ * <h4>Parameters</h4>
```

Comment:
Haha, honestly I'm not that concerned, but this is functionality we have scattered across the library. I was just thinking it might be nice to add support for it on the docs site so that we don't have to bake html into the JSDoc.